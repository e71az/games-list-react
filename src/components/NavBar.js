import { Link } from 'react-router-dom';

const NavBar = () => (
  <nav data-testid="navbar" className="col-12 navbar d-flex justify-content-center">
    <Link to="/home">flavourbomb</Link>
  </nav>
);

export default NavBar;
