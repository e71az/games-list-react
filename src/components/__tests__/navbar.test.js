import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar';

afterEach(() => {
  cleanup();
});

describe('NavBar component tests', () => {
  // This render is for normal testing
  render(
    <Router>
      <NavBar />
    </Router>,
  );

  // This renderer is for snapshot testing
  const tree = renderer
    .create(
      <Router>
        <NavBar />
      </Router>,
    )
    .toJSON();

  const navbarElement = screen.getByTestId('navbar');

  test('Component should Render', () => {
    expect(navbarElement).toBeInTheDocument();
  });
  test('Component should contain flavourbomb string', () => {
    expect(navbarElement).toHaveTextContent('flavourbomb');
  });

  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
