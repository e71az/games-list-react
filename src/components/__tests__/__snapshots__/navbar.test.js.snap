// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component tests Matches snapshot 1`] = `
<nav
  className="col-12 navbar d-flex justify-content-center"
  data-testid="navbar"
>
  <a
    href="/home"
    onClick={[Function]}
  >
    flavourbomb
  </a>
</nav>
`;
